src/base/utils.js

```diff
@@ -195,23 +196,31 @@ export function isNumber(value) {
 }
 
 export function currentScriptUrl() {
```

Comment:
Need to figure out how to do this before this. It needs to be called in the same tick as the script being loaded, but can't be called before the developer has a chance to set their `window` :/